CDCgov/prime-reportstream
# Filenames should prefix with none when translation schema is not present

This PR fixes the issue where FHIR filenames were being formed incorrectly. The standard filename format is `{translationSchema}-{reportId}-{dateTime}.{fileType}`. Because most (all?) FHIR receivers do not utilize a translation schema, this was being left as a blank string and formatting the filename as `-{reportId}-{dateTime}.{fileType}`. 

Regression tests/other scenarios that were considered:
- Topics with `isSendOriginal` set to true. 
  - These end up with a filename format of `None-{reportId}-{dateTime}.{fileType}`. This is due to where the SchemaName is set in Report.kt and a comment was added to highlight where this is happening.
  - Alternatively, reports with this topic setting can provide a `payloadname` when being submitted. This will result in an external name added to the file and this scenario **does not** use the logic in FileNameTemplate.kt. This scenario is [tested elsewhere](https://github.com/CDCgov/prime-reportstream/blob/2569a589d330c300477bbbff96694335dae2e84d/prime-router/src/test/kotlin/ReportTests.kt#L1071).
- Alternative filename template formats.
  - We have a [yml for defining alternative filename formats](https://github.com/CDCgov/prime-reportstream/blob/2569a589d330c300477bbbff96694335dae2e84d/prime-router/metadata/file_name_templates/file-name-templates.yml).
  - This fix only touches filename formats which utilize the `SchemaBaseName()` element. Filename templates with this element are `standard` and `standard_secondary`.
  - **Test have been added to cover these scenarios**.

Test Steps:
1. (_optional_) To reproduce the issue 
   1. Checkout master and run `./gradlew testEnd2EndUP`
   2. Look over the history endpoint results for the tests. You will notice a few things:
      1. Reports routed to receivers with no translation schema (can be verified by checking organizations.yml) will have the incorrect formatting of `-{reportId}-{dateTime}.{fileType}`. This will be the case for all receivers setup to receiver FHIR for this test.
      2. Reports for ELIMS (the isSendOriginal set to true scenario) will be formatted as `None-{reportId}-{dateTime}.{fileType}`.
      3. Reports routed to receivers with translation schemas listed will be properly formatted as `{translationSchema}-{reportId}-{dateTime}.{fileType}`. This will be the case for all receivers that receive HL7 in this test.
2. To verify the change
   1. Checkout the branch and run `./gradlew testEnd2EndUP`
   2. Look over the history endpoint results for the tests.
      1. Reports routed to FHIR receivers should now be prefixed with `none` and formatted as `none-{reportId}-{dateTime}.fhir`.
         1. The reason for lowercase is that the standard filename template is set to lowercase.
      2. Reports for ELIMS should be unchanged and will be formatted as `None-{reportId}-{dateTime}.hl7`.
      3. Reports routed to HL7 receivers should be unchanged and will be formatted as `{translationSchema}-{reportId}-{dateTime}.hl7`.

## Changes
- When a filename is generated for a receiver with no SchemaBaseName, this will be populated as `None` instead of an empty string.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #14455
